File: src/server/bo/BusinessObject.py
# Info: Der Großteil aus diesem Code wurde aus dem "Bankprojekt" übernommen

from abc import ABC


class BusinessObject(ABC):
    def __init__(self):
        self.id = 0

    def set_id(self, new_id):
        self.id = new_id

    def get_id(self):
        return self.id

File: frontend/eatsmarter/src/api/BusinessObject.js
/** Info: Der Großteil aus diesem Code wurde vom "Bankprojekt" übernommen  **/

/**
 * Basisklasse für alle BusinessObjekte mit Id per default
 */

export default class BusinessObject{
    // Null Constructor
    constructor() {
        this.id = 0;
    }

    // Id für das BusinessObjekt setzen
    setID(aId){
        this.id = aId;
    }

    // Id des BusinessObjekts ausgeben
    getID(){
        return this.id;
    }

    // Objekt als String ausgeben - Nützlich fürs debugging
    toString(){
        let result = "";
        for (var prop in this){
            result += prop + ": " + this[prop] + " ";
        }
        return result;
    }
}